2020-02-09 Ryan Brennan <rpbrennan@g.ucla.edu>

    * timespec_cmp
    changed subtraction of mtimes to conditional comparisons and pre-defined
    return values to prevent overflow

--- lib/timespec.h	2020-02-09 16:25:06.736713000 -0800
+++ lib/timespec.h	2020-02-09 16:25:59.467007000 -0800
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec) return 1;
+  else if (a.tv_sec < b.tv_sec) return -1;
+  else return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
